### PR TITLE
Generate a single warning per unknown header. Lock access to subscrib…

### PR DESCRIPTION
**Pull request by accelerated**
### Changes
* Generate a single warning per unknown header. 
* Lock access to subscribing/unsubscribing consumers in case of slight but potential race-condition in case the application calls subscribe/unsubscribe while there's an ongoing rebalance callback executing.

Signed-off-by: Alexander Damian <adamian@bloomberg.net>

### CODE REVIEW

File: corokafka/impl/corokafka_consumer_manager_impl.cpp [hunk 1]
```diff
@@ -961,33 +974,46 @@ ConsumerManagerImpl::deserializeMessage(ConsumerTopicEntry& entry,
...
-            }
-        }
-        catch (const std::exception& ex) {
+        auto deserIter = deserializer._headerDeserializers.find(header.get_name());
+        if (deserIter == deserializer._headerDeserializers.end()) {
+            std::ostringstream oss;
+            oss << "Found unknown header [" << header.get_name() << "] in topic " << toppar.get_topic();
             if (entry._skipUnknownHeaders) {
-                report(entry, cppkafka::LogLevel::LogWarning, {}, ex.what(), &kafkaMessage);
+                //Check if we have already warned about this unknown header
```
Comment (arosenzweig3) [inline corokafka/impl/corokafka_consumer_manager_impl.cpp:982]: I'm rather ambivalent about this part of the change; it's nice to not have the spam/spew in the logs, but that also decreases the odds that the library client will notice this error is coming up.
Unlike in some (entirely internal) libraries where we control the logging directly, we don't have a known place where the logs should be, which makes it hard for us to set up a ticket for the client.

Maybe instead of logging each unknown header once, we try to log all unknown headers for a given message once & together? i.e. "Found unknown header(s) [a, b] in topic TOPIC".

Basically my concern is that when skip unknown headers is on (which it is by default), we're doing work to reduce spam in the log, but we may be reducing it so much that clients may not even notice the skipped header, which may mask bugs in their configuration (there's no difference between an entirely unconfigured header, and a mis-named header).

That's my two cents or so, but I'm comfortable being outvoted here by other maintainers.

File: corokafka/corokafka_consumer_topic_entry.h [hunk 2]
```diff
@@ -116,6 +116,9 @@ struct ConsumerTopicEntry {
     ThrottleControl                 _throttleControl;
     bool                            _preserveMessageOrder{false};
     IoTracker                       _ioTracker;
+    std::set<std::string>           _skippedHeaders;
+    quantum::ReadWriteMutex         _skippedHeadersRWMutex;
+    quantum::Mutex                  _subscriptionMutex;
```
[the remark this answers is not in the document]
Comment (accelerated) [reply corokafka/corokafka_consumer_topic_entry.h:121]: Yes agreed, ideally these would be put under a sub-struct.
